qui/tray/gross_gtk3_bug_workaround.py

```diff
@@ -0,0 +1,120 @@
+        self.keep_going = False
+        self.runtime_cb = None
+        conn = self.conn = xcffib.connect()
+
+        setup = conn.get_setup()
+        if setup.roots_len != 1:
+            raise RuntimeError(f"X server has {setup.roots_len} screens, this is not supported")
+        screen, = setup.roots
+        # This is not guaranteed to work, but assume it will.
+        depth_32, = (depth for depth in screen.allowed_depths if depth.depth == 32)
+        self.window_id = p = conn.generate_id()
+        proto = self.proto = xcffib.xproto.xprotoExtension(conn)
+        assert screen.width_in_pixels > 0
+        assert screen.height_in_pixels > 0
+        fullscreen = "_NET_WM_STATE_FULLSCREEN"
+        wm_state_fullscreen_cookie = proto.InternAtom(only_if_exists=False,
+                                                      name_len=len(fullscreen),
+                                                      name=fullscreen,
+                                                      is_checked=True)
+        cookie1 = proto.CreateWindow(depth=0,
+                                     parent=screen.root,
+                                     x=0,
+                                     y=0,
+                                     wid=p,
+                                     width=screen.width_in_pixels,
+                                     height=screen.height_in_pixels,
+                                     border_width=0,
+                                     _class=xcffib.xproto.WindowClass.InputOnly,
+                                     visual=depth_32.visuals[0].visual_id,
+                                     value_mask=xcffib.xproto.CW.OverrideRedirect|xcffib.xproto.CW.EventMask,
+                                     value_list=[0,xcffib.xproto.EventMask.ButtonPress],
+                                     is_checked=True)
+        wm_state_fullscreen = wm_state_fullscreen_cookie.reply().atom
+        cookie2 = proto.ChangeProperty(mode=xcffib.xproto.PropMode.Replace,
+                                       window=p,
+                                       property=wm_state_fullscreen,
+                                       type=xcffib.xproto.Atom.ATOM,
+                                       format=32,
+                                       data_len=1,
+                                       data=[wm_state_fullscreen],
+                                       is_checked=True)
+        self.cookies = deque([cookie1, cookie2])
+        source = GLib.unix_fd_source_new(conn.get_file_descriptor(),
+                                         GLib.IOCondition(GLib.IO_IN|GLib.IO_HUP|GLib.IO_PRI))
+        GObject.source_set_closure(source, self.source_callback)
+        main_loop = GLib.main_context_ref_thread_default()
+        assert main_loop is not None
+        source_id = source.attach(main_loop)
+        self.cb()
+    def show_for_widget(self, widget) -> None:
+        widget.connect("unmap-event", lambda _unused: self.hide())
```

Comment:
At least based on the name, this event looks to be x11 specific. Should it check then if the main application is really using x11 backend?